src/OpenSpaceToolkit/Astrodynamics/Trajectory/Models/Tabulated.cpp

```diff
@@ -25,12 +25,61 @@ namespace models
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-                                Tabulated::Tabulated                        (   const   Array<State>&               aStateArray                                 )
+                                Tabulated::Tabulated                        (   const   Array<State>&               aStateArray,
+                                                                                const   InterpolationType&          anInterpolationType                                 )
                                 :   Model(),
-                                    states_(aStateArray),
-                                    stateIndex_(0)
+                                    interpolationType_(anInterpolationType)
 {
 
+    using ostk::math::curvefitting::interp::CubicSpline ;
+    using ostk::math::curvefitting::interp::BarycentricRational ;
+    using ostk::math::curvefitting::interp::LinearInterpolator ;
+
+    if (aStateArray.isEmpty())
+    {
+        return ;
+    }
```

Comment:
Might be worth checking that at least 2 elements are provided? I think initializing a Linear Interpolator instant with a single state would throw an error, nah? Let me know if I am missing sth

Reply:
Yeah that's a great point.